File: 3100-water-bottles-ii/3100-water-bottles-ii.java
class Solution{ 
    public int maxBottlesDrunk(int numBottles, int numExchange) {
        int emptyBottles=numBottles;
        while(numExchange<=numBottles){
            numBottles-=numExchange;
            numExchange++;
            numBottles++;
            emptyBottles++;
        }
        return emptyBottles;
    }
}

File: 3100-water-bottles-ii/3100-water-bottles-ii.py
class Solution(object):
    def maxBottlesDrunk(self, numBottles, numExchange):
        emptyBottle=numBottles
        while numBottles >= numExchange:
            numBottles-=numExchange
            numExchange+=1
            numBottles+=1
            emptyBottle+=1
        return emptyBottle
        